marshal/gogoProtoMarshalizer.go

```diff
@@ -18,20 +13,25 @@ type GogoProtoMarshalizer struct {
 // Marshal does the actual serialization of an object
 // The object to be serialized must implement the gogoProtoObj interface
 func (x *GogoProtoMarshalizer) Marshal(obj interface{}) ([]byte, error) {
-	if msg, ok := obj.(gogoProtoObj); ok {
+	if msg, ok := obj.(data.GogoProtoObj); ok {
 		return msg.Marshal()
 	}
-	return nil, ErrMarshallingProto
+	return nil, fmt.Errorf("%T, %w", obj, ErrMarshallingProto)
 }
 
 // Unmarshal does the actual deserialization of an object
 // The object to be deserialized must implement the gogoProtoObj interface
 func (x *GogoProtoMarshalizer) Unmarshal(obj interface{}, buff []byte) error {
-	if msg, ok := obj.(gogoProtoObj); ok {
+	if msg, ok := obj.(data.GogoProtoObj); ok {
 		msg.Reset()
-		return msg.Unmarshal(buff)
+		err := msg.Unmarshal(buff)
+		if err != nil {
+			panic(err)
```

Comment:
Is it necessary to panic?
What if the deserialization is done on a received buffer?
I think the panic should be removed.

Reply:
It was helpful in the integration ...., removed